### ISSUE TITLE
greenkeeper-lockfile never gets triggered

### ISSUE THREAD

**Issue by sarneeh** [greenkeeperio/greenkeeper-lockfile#208]
I don't get it how this package should work. This doesn't make any sense.

I trigger a TravisCI build on every new pull request. Important note: **pull request** , not **push**. Every new pull request on Travis is getting built by a merge of the default branch (master) and the pull request one. Travis **tags the build as being built on master**, not on some other branch. This results in a `'master' is not a Greenkeeper branch` message and makes `greenkeeper-lockfile` completely useless, as it will never get triggered - on PRs branch is marked as `master`, on merge and finally push to `master` it's also obviously marked as `master`.

Example PR: https://github.com/sarneeh/reaptcha/pull/57
Example build: https://travis-ci.org/sarneeh/reaptcha/builds/422454239

Am I missing something or there's something wrong here? 😄

**Comment by kylemh** [comment]
I'm having the same issue. I followed documentation to the letter.

Example PR: https://github.com/OperationCode/front-end/pull/98
Travis file: https://github.com/OperationCode/front-end/blob/master/.travis.yml

**Comment by Kornil** [comment]
Same for me, tried rewriting the travis config for a week to make it work but it never triggered once, not even with the simplest config.
It either ends with 
- `'master' is not a Greenkeeper branch` (on pretty much everything that is not the first commit of the greenkeeper PR)
- `Only running on first push of a new branch` on the first push. Even tho greenkeeper doesn't even make 1 commit PRs, it usually adds multiple for every package.

see https://github.com/Kornil/personal-website/pull/10 on "details" in checks you'll see both errors at the end of the logs.

**Comment by elwayman02** [comment]
My formerly working config is also exhibiting the `master is not a Greenkeeper branch` error now: https://travis-ci.org/elwayman02/ember-interactivity/jobs/422262174

**Comment by Realtin** [comment]
The `'master' is not a Greenkeeper branch` message will be displayed on the travis `pr` run. Greenkeeper-lockfile only runs on the `push` (branch status). Having a CI branch status is a requirement for Greenkeeper.

@sarneeh greenkeeper-lockfile only runs on branches that are created by [greenkeeper](https://github.com/greenkeeperio/greenkeeper)

@kylemh On the [`push` run](https://travis-ci.org/OperationCode/front-end/builds/422836770?utm_source=github_status&utm_medium=notification) it says `No dependency changed` which is probably because you run `yarn global add greenkeeper-lockfile@1`but monorepos are only supported with `greenkeeper-lockfile@2`

@Kornil similar problem here. Since this was a monorepo release it might require greenkeeper-lockfile@2. version 2 is tagged as `next`release and not `latest` therefore has to be explicitly installed with the `@2` 

@elwayman02 this is a `pr` travis run which bases of the master branch and therefore greenkeeper-lockfile does not run. I checked a [push/branch run](https://travis-ci.org/elwayman02/ember-interactivity/jobs/422261334) and there the lockfile was updated.

_We are currently working on supporting Lockfile updates natively within Greenkeeper._  ✨ 🌴
